src/compiler/utilities.ts

```diff
@@ -438,6 +525,46 @@ module ts {
         }
     }
 
+    export function forEachYieldExpression(body: Block, visitor: (expr: YieldExpression) => void): void {

```

Comment:
The reason I used void is because yield expressions can be nested. If this could be used to return something for each yield expression I find, then it's not clear in what order the results should be returned (pre-order vs post-order). I avoided the question by making it void. This might not be a good argument. But anyone can change it if they need to return something else.

For the forEachReturnStatement case, we actually do use it to return a value sometimes (I think a boolean), that's why I left it alone instead of converting it to void.